opensquirrel/decomposer/aba_decomposer.py

```diff
@@ -0,0 +1,142 @@
+from __future__ import annotations
+
+import math
+from abc import ABC, abstractmethod
+from collections.abc import Callable, Iterable
+
+import numpy as np
+
+from opensquirrel.common import ATOL
+from opensquirrel.decomposer.general_decomposer import Decomposer
+from opensquirrel.default_gates import Rx, Ry, Rz
+from opensquirrel.ir import Axis, AxisLike, BlochSphereRotation, Float, Gate, Qubit
+from opensquirrel.utils.identity_filter import filter_out_identities
+
+
+class ABADecomposer(Decomposer, ABC):
+    @property
+    @abstractmethod
+    def ra(self) -> Callable[..., BlochSphereRotation]: ...
+
+    @property
+    @abstractmethod
+    def rb(self) -> Callable[..., BlochSphereRotation]: ...
+
+    _gate_list: list[str] = [Rx.__name__, Ry.__name__, Rz.__name__]
+
+    def __init__(self) -> None:
+        self.index_a = self._gate_list.index(self.ra.__dict__["__wrapped__"].__name__)
+        self.index_b = self._gate_list.index(self.rb.__dict__["__wrapped__"].__name__)
```

Comment:
I find this quite obscure. Wouldn't we be better off with the way we implemented it initially?

```suggestion
    _gate_list: list[Callable[..., BlockSpereRotation]] = [Rx, Ry, Rz]

    def __init__(self) -> None:
        self.index_a = self._gate_list.index(self.ra)
        self.index_b = self._gate_list.index(self.rb)
```

Reply:
I've created a PR to fix the ABADecomposer and the imports of ZYZDecomposer.

The error messages you mention come from the fact that all the ABADecomposer subclasses have to implement all the ABADecomposer abstract methods, including the properties `ra` and `rb`.